apl/tabular-operators/lookup-operator.mdx

```diff
@@ -0,0 +1,132 @@
+---
+title: lookup
+description: 'This page explains how to use the lookup operator in APL.'
+---
+
+## Introduction
+
+The `lookup` operator extends a primary dataset with a lookup dataset based on a specified key column. It retrieves matching rows from the lookup dataset and appends relevant fields to the primary dataset. You can use `lookup` for enriching event data, adding contextual information, or correlating logs with reference tables.
+
+The `lookup` operator is useful when:
+
+- You need to enrich log events with additional metadata, such as mapping user IDs to user profiles.
+- You want to correlate security logs with threat intelligence feeds.
+- You need to extend OpenTelemetry traces with supplementary details, such as service dependencies.
+
+## For users of other query languages
+
+If you come from other query languages, this section explains how to adjust your existing queries to achieve the same results in APL.
+
+<AccordionGroup>
+<Accordion title="Splunk SPL users">
+
+In Splunk SPL, the `lookup` command performs a similar function by enriching event data with fields from an external lookup table.
+
+<CodeGroup>
+```sql Splunk example
+index=web_logs | lookup geoip_lookup ip AS client_ip OUTPUT city, country
+```
+
+```kusto APL equivalent
+['sample-http-logs']
+| lookup kind=inner ['geoip_lookup'] on id
```

Comment:
GeoIP lookup might not be the best example, as you'd use the function in that case. Perhaps something like "PortNameLookup" to look at common services that are connecting to your server?